## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 416177 --package deno`
Commit: `cc59cdd4c68ed164c2859448f51450e7cfb27aae`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>deno</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>deno</summary>
<pre>error: linking with `/nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/cc` failed: exit status: 1
  |
  = note:  "/nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/cc" "-m64" "/build/rustcmwPckM/symbols.o" "<1 object files omitted>" "-Wl,--as-needed" "-Wl,-Bstatic" "/build/rustcmwPckM/{libzstd_sys-066512c6f19b51fa.rlib,libdprint_plugin_jupyter-906a08cdb5b51963.rlib,libdprint_plugin_json-1773a42e650662b0.rlib,libdprint_plugin_markdown-a1bee736e2b532d7.rlib,libpulldown_cmark-2eed28cffa22737e.rlib,libunicase-c2aa3203aa5bfe77.rlib,libdprint_plugin_typescript-ac3a4ab91d9905fd.rlib,libdprint_core-1d8293d5580ed10b.rlib,libdeno_doc-827f0cbd1c39ba36.rlib,libhtml_escape-07eb3b480d025272.rlib,libutf8_width-697d56adcbf4ab2b.rlib,libcomrak-bb43990025a610ed.rlib,libcaseless-9a167d27bbbdee9e.rlib,libunicode_normalization-c003ade2c23fa3e8.rlib,libunicode_categories-5212042732ac029f.rlib,libentities-4a7eb35dcc5a1bd8.rlib,libhandlebars-71cc32b1a10851b2.rlib,libpest-15d69bd8da8621ea.rlib,libucd_trie-10e5b4b9b248fcbc.rlib,libheck-7822b7ffda3891dd.rlib,libderive_builder-cd23eced670ee7d5.rlib,libthiserror-1b6138562a5f9a09.rlib,libtinyvec-795ac6c652069e52.rlib,libtinyvec_macros-ffd3466e6bbfe33a.rlib,libring-eb75864ec9254488.rlib,liblcms2_sys-17e1839f83b12751.rlib,libzune_jpeg-8f2ca45d32d71276.rlib,libzune_core-646d359e6d6df3ad.rlib,liblibsqlite3_sys-1b39ba74873d88db.rlib,libpathdiff-1e618bd5b5c610bc.rlib,liblibz_sys-f8583b325ab1c7ae.rlib,libdeno_cache_dir-3de9b12d980837de.rlib,libbase32-fb0db485061598bd.rlib,libchrono-13e3811343d66f31.rlib,libcache_control-09eaed4cd1950b95.rlib,libbase64-4b8026056b645a05.rlib,libhttp-f9e6a15b05c91532.rlib,libfnv-e3e47113262bae7c.rlib,libdeno_graph-22a98888dac0af36.rlib,libtwox_hash-f8ab64d5e7a956d2.rlib,libsha2-618074d1caf4455c.rlib,libdeno_ast-a7fdd87e6afc4fc4.rlib,libswc_ecma_transforms_typescript-829febda3892903c.rlib,libswc_ecma_transforms_react-fe258acbf67687d6.rlib,libsha1-b33ddb7b06dee7b8.rlib,libcpufeatures-4bf2de90faaca577.rlib,libdigest-e3107a9be50379e6.rlib,libsubtle-41cd5480e0294a0e.rlib,libconst_oid-c58d1e6ead3c0363.rlib,libblock_buffer-2ba3ec7ba9c80bf2.rlib,libcrypto_common-a887fd7ed8ed66f8.rlib,libgeneric_array-941a742557d16798.rlib,libtypenum-b37aaf6844e7384a.rlib,libzeroize-1544223d5af5eb0d.rlib,libswc_config-eb13d2578378056f.rlib,libswc_ecma_transforms_proposal-cd200ee0d22e6d51.rlib,libswc_ecma_transforms_classes-0c0643a7f82b8144.rlib,libswc_bundler-a821057ef53bb884.rlib,libradix_fmt-d5426810c9004e47.rlib,libswc_ecma_loader-ae3c0d3db5809715.rlib,libcrc-f573d80ee6ea4688.rlib,libcrc_catalog-218434e91eb7d3a3.rlib,libswc_ecma_codegen-3c13e58c1420a676.rlib,libswc_allocator-bf777c1ee3d97002.rlib,libascii-2c55f1b8eaa1e1f0.rlib,libcompact_str-217ad8b266d0c96f.rlib,libcastaway-928088f14df0b353.rlib,libswc_ecma_transforms_optimization-26e0976f62c8d3c8.rlib,libdashmap-377a9c9bb96864e6.rlib,librelative_path-a002da83d12c3754.rlib,libswc_graph_analyzer-39c42641126424bb.rlib,libswc_ecma_transforms_base-53bc5043853f22e8.rlib,libpetgraph-0ebca651613b42d6.rlib,libfixedbitset-ac89e7aeaa12f31a.rlib,libswc_ecma_utils-6ad732963e323a9b.rlib,libryu_js-1ef204cb8dd47824.rlib,libpar_core-2c71f93a429058c3.rlib,libswc_ecma_visit-ebbd125249b3372c.rlib,libdprint_swc_ext-f47a9c6359375b66.rlib,libbumpalo-92748bbd32bdac45.rlib,libswc_ecma_parser-b92d990fa7a81242.rlib,libstacker-a18777a6406ad916.rlib,libpsm-b7edec808615a622.rlib,libtyped_arena-8805a1bb817efb3d.rlib,libswc_ecma_lexer-6d16b877d7e9ab7e.rlib,libsmartstring-541f4767df6f8d95.rlib,libarrayvec-edd747c7e856e8da.rlib,libswc_ecma_ast-cb918eb3e6fdc785.rlib,libtext_lines-874c94c2c90c8052.rlib,libswc_common-bd9b4408be75a488.rlib,libbetter_scoped_tls-a2dfe13f7933ca5e.rlib,libscoped_tls-ec6ea8d0e81e02c7.rlib,libunicode_width-67ca3e52e726e4a7.rlib,libswc_visit-3db428e1b1240eb3.rlib,libeither-a9fbfb4e25862092.rlib,libtracing-d0f7d3f968736eaf.rlib,libtracing_core-330e740533cf0723.rlib,libswc_atoms-9eea8da435fe00da.rlib,libhstr-28bca3a0a4a37200.rlib,libhashbrown-8dddc54a3df79cf8.rlib,libahash-92ab5f33d9e487df.rlib,libzerocopy-764cf3ca31d749ed.rlib,libbyteorder-08aa7bd22e0b4cf7.rlib,libtriomphe-f6ab39b5c1af3c35.rlib,libdebug_unreachable-d0986de9a3bb9f30.rlib,libbase64-22dfba646e78b5f7.rlib,libunicode_width-14456af17b18f5b1.rlib,libdeno_terminal-e1dbdef3da53ff69.rlib,libtermcolor-cf2c24ee80f4737e.rlib,libdeno_media_type-d49fc946cd2bf29c.rlib,libencoding_rs-946eec01191849bc.rlib,libdata_url-bb8e1c90818642f1.rlib,libstatic_assertions-c059fe85cee1a6ce.rlib,libanyhow-91d3c49e3d319e40.rlib,libsourcemap-d959ca586ca1a917.rlib,libbase64_simd-d5a268654c85d51d.rlib,liboutref-a3fbaa851f99d6bc.rlib,libvsimd-2b0867b0d50ce42f.rlib,libunicode_id_start-fb97bbc72e434c15.rlib,libdata_encoding-af18cecda4b05c30.rlib,libif_chain-5d7352ea45440b2a.rlib,libbitvec-ea43d57d17c4cf21.rlib,libwyz-0ef13b9e9c98ffdc.rlib,libtap-0adc8e2d506c2628.rlib,libradium-0a2f6c865d7644a1.rlib,libfunty-3ecac97c7a34717f.rlib,librustc_hash-f0aef0d3390e3d4c.rlib,libdebugid-1c232b29253810dd.rlib,libuuid-8216ab92d3693bea.rlib,libsha1_smol-12350c2b2629ba2e.rlib,libdeno_unsync-325a53b026931269.rlib,libwasm_dep_analyzer-ef1ecf17bcf49e9b.rlib,libnum_bigint-9899143598f950a7.rlib,librand-2f6ad423d04cc55c.rlib,librand_chacha-b38eecea18e8a89e.rlib,libppv_lite86-61b59063fd29c990.rlib,librand_core-4c99070881193b03.rlib,libnum_integer-76ca14c352d8b6a4.rlib,libnum_traits-29da208a587d1cf5.rlib,libv8-85c62ac54f09abd9.rlib,libfutures-3a49342b9e703b1a.rlib,libfutures_executor-702829a1047a719f.rlib,libfutures_util-186285552f0f8624.rlib,libfutures_io-4985afa183cf970a.rlib,libslab-2ee383143333b723.rlib,libfutures_channel-2f747bf18ca43a40.rlib,libfutures_sink-260afaeec588ed8d.rlib,libfutures_task-39e7f1c573d45b57.rlib,libpin_utils-6fc538a2d13aac32.rlib,libfutures_core-49b1b8152e76cbee.rlib,liblazy_static-7e705ea2f25bad1c.rlib,libspin-59e89dd11424bda7.rlib,libregex-e2c8ddcb8990b6ad.rlib,libregex_automata-38531ad09f7d0660.rlib,libregex_syntax-c87f254115487dd0.rlib,libaho_corasick-0305a0bddec40583.rlib,libphf-0b664f30c6930da2.rlib,libphf_shared-e3d4b4526702a21d.rlib,libsiphasher-2186c221641cd73e.rlib,libdeno_semver-ed0912e91a3853a1.rlib,libmonch-7cfdbf3822f94426.rlib,libonce_cell-a555cbd81261d2bf.rlib,libcapacity_builder-15c4e23169454b10.rlib,libhipstr-49cc555969aac813.rlib,libserde_bytes-d937beac01971bbd.rlib,libsptr-6966c54bd0009e51.rlib,libecow-89ba71a293519e5b.rlib,libjsonc_parser-010ba8303da6bddc.rlib,libimport_map-db15fb1704f7f000.rlib,libdeno_path_util-7b00421a01cdceb4.rlib,libthiserror-0b1335cb7dc56718.rlib,libsys_traits-ba4e0ac3642580a2.rlib,libgetrandom-2b892424bc7c5448.rlib,libfiletime-f22d0ff0f1e29dff.rlib,libdeno_error-3183b2b3bfaf5fa8.rlib,libtokio-bb4735846093db6d.rlib,libsignal_hook_registry-125ff2ded6cd4e9b.rlib,libnum_cpus-50bf23aedd3a2fea.rlib,libsocket2-ecbe8909ca3ca743.rlib,libbytes-1b37d04637cacd92.rlib,libmio-e7178d669676d850.rlib,liblog-6c2e5a0a36ab1250.rlib,libparking_lot-0dfe865752717637.rlib,libparking_lot_core-3022fd697568a648.rlib,libcfg_if-eedd0609c1a50b93.rlib,liblock_api-457a3e753fb93f67.rlib,libscopeguard-4e07fc37fc41c2d5.rlib,libpin_project_lite-3a6b161298f0d156.rlib,liburl-66dbe98377c10417.rlib,libidna-6f354f4a69e3f33d.rlib,libidna_adapter-6480b47158871b01.rlib,libicu_normalizer-8629e89fd41dfcfc.rlib,libicu_normalizer_data-3045e09ef68a6dc2.rlib,libwrite16-dc8908449969f95d.rlib,libutf8_iter-8047c41b65a9bdab.rlib,libutf16_iter-0588c82588d71f8d.rlib,libsmallvec-67ef72a96c448454.rlib,libicu_properties-7c6de2ceed6d148a.rlib,libicu_properties_data-12ef77b2054d1ebb.rlib,libicu_locid_transform-75b7c43168a87c9f.rlib,libicu_locid_transform_data-373ef85c8ef2badb.rlib,libicu_collections-3de11534baa7d9c4.rlib,libicu_provider-ef9b7aa88546a5ad.rlib,libicu_locid-707bd62ab261bc7b.rlib,liblitemap-84af18206cce8852.rlib,libtinystr-1f1472c7b103e34f.rlib,libzerovec-80ec9398821325a2.rlib,libwriteable-1b4151d89abe0fa5.rlib,libyoke-f086355d4223d7ea.rlib,libzerofrom-2cd91f4356446f7d.rlib,libstable_deref_trait-7f247a68a37ce658.rlib,libform_urlencoded-65b70eca8154bf0d.rlib,libpercent_encoding-d36f69503dfe1b9b.rlib,liblibc-9c45daeb870b3835.rlib,libserde_json-e4e412dade0c19b0.rlib,libmemchr-32356740144b10a8.rlib,libitoa-e2b8eb9c65b06626.rlib,libryu-2b053edbae5cbfc6.rlib,libindexmap-f973589167412cc0.rlib,libhashbrown-4623852d466ca933.rlib,libfoldhash-39da879418f9ec59.rlib,libequivalent-f9a663c99f0e9ea1.rlib,liballocator_api2-a5f1350d5f32ae80.rlib,libbitflags-7ee986c1bd6754bf.rlib,libserde-143d9db5eb59e5af.rlib}.rlib" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib/{libcompiler_builtins-*}.rlib" "-Wl,-Bdynamic" "-ldl" "-ldl" "-lffi" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-L" "/build/rustcmwPckM/raw-dylibs" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/build/source/target/x86_64-unknown-linux-gnu/release/build/psm-f8979f39db373a53/out" "-L" "/build/source/target/x86_64-unknown-linux-gnu/release/gn_out/obj" "-L" "/build/source/target/x86_64-unknown-linux-gnu/release/build/ring-bc036d410e2a5e01/out" "-L" "/build/source/target/x86_64-unknown-linux-gnu/release/build/libz-sys-665c8becf7a8cdf1/out/lib" "-L" "/build/source/target/x86_64-unknown-linux-gnu/release/build/libz-sys-665c8becf7a8cdf1/out/lib64" "-L" "/build/source/target/x86_64-unknown-linux-gnu/release/build/libsqlite3-sys-4cd5f9eecd41c828/out" "-L" "/build/source/target/x86_64-unknown-linux-gnu/release/build/lcms2-sys-dc4e484618754028/out" "-L" "/build/source/target/x86_64-unknown-linux-gnu/release/build/zstd-sys-25e1c76f0c3f5726/out" "-L" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/deps/deno-f386821949ca830c" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs" "-Wl,--export-dynamic-symbol-list=/build/source/ext/napi/generated_symbol_exports_list_linux.def" "/build/source/target/x86_64-unknown-linux-gnu/release/deps/deno_audit_data.o" "-Wl,--undefined=AUDITABLE_VERSION_INFO"
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: /nix/store/z3za8hfc24wb117s50p8b10agjkgm039-binutils-2.44/bin/ld: /build/source/target/x86_64-unknown-linux-gnu/release/deps/deno-f386821949ca830c.zip-4afc1a674b29607a.zip.66a1f30767e2a9e5-cgu.0.rcgu.o.rcgu.o: warning: relocation against `_ZN4ring6pbkdf216PBKDF2_HMAC_SHA117h371ea527a9bc3fc1E' in read-only section `.text._ZN5tokio7runtime4task3raw4poll17hde790d14ded0ade1E'
          /nix/store/z3za8hfc24wb117s50p8b10agjkgm039-binutils-2.44/bin/ld: /build/source/target/x86_64-unknown-linux-gnu/release/deps/deno-f386821949ca830c.zip-4afc1a674b29607a.zip.66a1f30767e2a9e5-cgu.0.rcgu.o.rcgu.o: in function `tokio::runtime::task::raw::poll':
          /build/rustc-1.87.0-src/library/core/src/option.rs:(.text._ZN5tokio7runtime4task3raw4poll17h5a3a55e56fe3edb8E+0x13b): undefined reference to `ring::pbkdf2::PBKDF2_HMAC_SHA1'
          /nix/store/z3za8hfc24wb117s50p8b10agjkgm039-binutils-2.44/bin/ld: /build/rustc-1.87.0-src/library/core/src/option.rs:(.text._ZN5tokio7runtime4task3raw4poll17h5a3a55e56fe3edb8E+0x23c): undefined reference to `ring::pbkdf2::PBKDF2_HMAC_SHA1'
          /nix/store/z3za8hfc24wb117s50p8b10agjkgm039-binutils-2.44/bin/ld: /build/source/target/x86_64-unknown-linux-gnu/release/deps/deno-f386821949ca830c.zip-4afc1a674b29607a.zip.66a1f30767e2a9e5-cgu.0.rcgu.o.rcgu.o: in function `tokio::runtime::task::raw::poll':
          /build/rustc-1.87.0-src/library/core/src/option.rs:(.text._ZN5tokio7runtime4task3raw4poll17h97845c76d6dd893cE+0x3a8): undefined reference to `ring::pbkdf2::PBKDF2_HMAC_SHA1'
          /nix/store/z3za8hfc24wb117s50p8b10agjkgm039-binutils-2.44/bin/ld: /build/source/target/x86_64-unknown-linux-gnu/release/deps/deno-f386821949ca830c.zip-4afc1a674b29607a.zip.66a1f30767e2a9e5-cgu.0.rcgu.o.rcgu.o: in function `tokio::runtime::task::raw::poll':
          /build/rustc-1.87.0-src/library/core/src/option.rs:(.text._ZN5tokio7runtime4task3raw4poll17hde790d14ded0ade1E+0x42e): undefined reference to `ring::pbkdf2::PBKDF2_HMAC_SHA1'
          /nix/store/z3za8hfc24wb117s50p8b10agjkgm039-binutils-2.44/bin/ld: warning: creating DT_TEXTREL in a PIE
          collect2: error: ld returned 1 exit status

  = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-link-lib)

error: could not compile `deno` (bin "deno") due to 1 previous error</pre>
</details>
</details>

I'll try to compile again to make sure it's not a fluke.

--- a/pkgs/by-name/de/deno/package.nix
+++ b/pkgs/by-name/de/deno/package.nix
@@ -3,6 +3,7 @@
   lib,
   callPackage,
   fetchFromGitHub,
+  fetchpatch,
   rustPlatform,
   cmake,
   yq,
@@ -21,6 +22,7 @@
   nodejs,
   git,
   python3,
+  esbuild,
 }:
 
 let
@@ -28,18 +30,26 @@ let
 in
 rustPlatform.buildRustPackage (finalAttrs: {
   pname = "deno";
-  version = "2.3.5";
+  version = "2.3.6";
 
   src = fetchFromGitHub {
     owner = "denoland";
     repo = "deno";
     tag = "v${finalAttrs.version}";
     fetchSubmodules = true; # required for tests
-    hash = "sha256-lu3r1v3iB9NIruooRrV9NawUnKqufqlYJQe+Aumgn8E=";
+    hash = "sha256-l3cWnv2cEmoeecYj38eMIlgqlRjDbtQuc6Q3DmOJoqE=";
   };
 
   useFetchCargoVendor = true;
-  cargoHash = "sha256-XJy7+cARYEX8tAPXLHJnEwXyZIwPaqhM7ZUzoem1Wo0=";
+  cargoHash = "sha256-alvn+d7XTYrw8KXw+k+++J3CsBwAUbQQlh24/EOEzwY=";
+  cargoPatches = [
+    (fetchpatch {
+      name = "fix-sigsegv-on-x86_64-unknown-linux-gnu-targets";
+      url = "https://github.com/denoland/deno/commit/400a9565c335b51d78c8909f4dbf1dbd4fb5e5d8.patch";
+      hash = "sha256-dTIw7P6sB6Esf+lSe/gc3cX54GkzLWF5X55yxP/QYoo=";
+      includes = [ "cli/Cargo.toml" ];
+    })
+  ];
 
   patches = [
     ./tests-replace-hardcoded-paths.patch
@@ -82,11 +92,42 @@ rustPlatform.buildRustPackage (finalAttrs: {
   # To avoid this we pre-download the file and export it via RUSTY_V8_ARCHIVE
   env.RUSTY_V8_ARCHIVE = librusty_v8;
 
-  preCheck = lib.optionalString stdenv.hostPlatform.isDarwin ''
-    # Unset the env var defined by bintools-wrapper because it triggers Deno's sandbox protection in some tests.
-    # ref: https://github.com/denoland/deno/pull/25271
-    unset LD_DYLD_PATH
-  '';
+  # Many tests depend on prebuilt binaries being present at `./third_party/prebuilt`.
+  # We provide nixpkgs binaries for these for all platforms, but the test runner itself only handles
+  # these four arch+platform combinations.
+  doCheck =
+    stdenv.hostPlatform.isDarwin
+    || (stdenv.hostPlatform.isLinux && (stdenv.hostPlatform.isAarch64 || stdenv.hostPlatform.isx86_64));
+
+  preCheck =
+    # Provide esbuild binary at `./third_party/prebuilt/` just like upstream:
+    # https://github.com/denoland/deno_third_party/tree/master/prebuilt
+    # https://github.com/denoland/deno/blob/main/tests/util/server/src/servers/npm_registry.rs#L402
+    let
+      platform =
+        if stdenv.hostPlatform.isLinux then
+          "linux64"
+        else if stdenv.hostPlatform.isDarwin then
+          "mac"
+        else
+          throw "Unsupported platform";
+      arch =
+        if stdenv.hostPlatform.isAarch64 then
+          "aarch64"
+        else if stdenv.hostPlatform.isx86_64 then
+          "x64"
+        else
+          throw "Unsupported architecture";
+    in
+    ''
+      mkdir -p ./third_party/prebuilt/${platform}
+      cp ${lib.getExe esbuild} ./third_party/prebuilt/${platform}/esbuild-${arch}
+    ''
+    + lib.optionalString stdenv.hostPlatform.isDarwin ''
+      # Unset the env var defined by bintools-wrapper because it triggers Deno's sandbox protection in some tests.
+      # ref: https://github.com/denoland/deno/pull/25271
+      unset LD_DYLD_PATH
+    '';
 
   cargoTestFlags = [
     "--lib" # unit tests
@@ -119,6 +160,8 @@ rustPlatform.buildRustPackage (finalAttrs: {
 
       # Flaky
       "--skip=init::init_subcommand_serve"
+      "--skip=serve::deno_serve_parallel"
+      "--skip=js_unit_tests::stat_test" # timing-sensitive
 
       # Test hangs, needs investigation
       "--skip=repl::pty_complete_imports_no_panic_empty_specifier"

--- a/pkgs/by-name/de/deno/package.nix
+++ b/pkgs/by-name/de/deno/package.nix
@@ -56,10 +56,20 @@ rustPlatform.buildRustPackage (finalAttrs: {
     ./tests-darwin-differences.patch
     ./tests-no-chown.patch
   ];
-  postPatch = ''
-    # Use patched nixpkgs libffi in order to fix https://github.com/libffi/libffi/pull/857
-    tomlq -ti '.workspace.dependencies.libffi = { "version": .workspace.dependencies.libffi, "features": ["system"] }' Cargo.toml
-  '';
+  postPatch =
+    ''
+      # Use patched nixpkgs libffi in order to fix https://github.com/libffi/libffi/pull/857
+      tomlq -ti '.workspace.dependencies.libffi = { "version": .workspace.dependencies.libffi, "features": ["system"] }' Cargo.toml
+    ''
+    +
+      lib.optionalString
+        (stdenv.hostPlatform.isLinux || (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86_64))
+        ''
+          # LTO crashes with the latest Rust + LLVM combination.
+          # https://github.com/rust-lang/rust/issues/141737
+          # TODO: remove this once LLVM is upgraded to 20.1.7
+          tomlq -ti '.profile.release.lto = false' Cargo.toml
+        '';
 
   buildInputs = [
     libffi

--- a/pkgs/by-name/de/deno/package.nix
+++ b/pkgs/by-name/de/deno/package.nix
@@ -162,9 +162,15 @@ rustPlatform.buildRustPackage (finalAttrs: {
       "--skip=init::init_subcommand_serve"
       "--skip=serve::deno_serve_parallel"
       "--skip=js_unit_tests::stat_test" # timing-sensitive
+      "--skip=repl::pty_complete_imports"
+      "--skip=repl::pty_complete_expression"
 
       # Test hangs, needs investigation
       "--skip=repl::pty_complete_imports_no_panic_empty_specifier"
+
+      # Use of VSOCK, might not be available on all platforms
+      "--skip=js_unit_tests::serve_test"
+      "--skip=js_unit_tests::fetch_test"
     ]
     ++ lib.optionals stdenv.hostPlatform.isDarwin [
       # Expects specific shared libraries from macOS to be linked

